src/main/java/net/snowflake/client/core/arrow/ArrowVectorConverterHelper.java

```diff
@@ -0,0 +1,219 @@
+package net.snowflake.client.core.arrow;
+
+import java.util.Map;
+import net.snowflake.client.core.DataConversionContext;
+import net.snowflake.client.core.SFBaseSession;
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.ErrorCode;
+import net.snowflake.client.jdbc.SnowflakeSQLException;
+import net.snowflake.client.jdbc.SnowflakeSQLLoggedException;
+import net.snowflake.client.jdbc.SnowflakeType;
+import net.snowflake.common.core.SqlState;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.complex.FixedSizeListVector;
+import org.apache.arrow.vector.complex.ListVector;
+import org.apache.arrow.vector.complex.MapVector;
+import org.apache.arrow.vector.complex.StructVector;
+import org.apache.arrow.vector.types.Types;
+
+@SnowflakeJdbcInternalApi
+public final class ArrowVectorConverterHelper {
+  private ArrowVectorConverterHelper() {}
```

Comment:
Depends on the convention I guess but a nit comment is that it may be a utility class instead of helper as there's no state to be stored.